File: frontend/src/types/template.ts
export interface MessageTemplate {
  id: number;
  template_name: string;
  delivery_channel: string;
  message_configuration: {
    subject: string;
    body: string;
  };
}

export interface MessageTemplateRequest {
  template_name: string;
  delivery_channel: string;
  message_configuration: {
    subject: string;
    body: string;
  };
}

File: backend/app/models/template.py
from sqlalchemy import Column, Integer, String
from sqlalchemy.dialects.postgresql import JSONB

from app.database import Base


class MessageTemplate(Base):
    __tablename__ = "message_templates"

    id = Column(Integer, primary_key=True, index=True)
    template_name = Column(String(100), nullable=False)
    delivery_channel = Column(String(50), nullable=False)
    message_configuration = Column(JSONB, nullable=False)